### PR TITLE
Allow users to switch between color and linetype aesthetics

### LINKED ISSUE #166: ggcuminc option to switch between showing multiple outcomes and strata as different linetypes and colors
**Is your feature request related to a problem? Please describe.**
ggcuminc defaults to show multiple outcomes as different linetypes and strata as different colors, but it is to my knowledge not possible to easily switch them, so outcomes will be shown as different colors and strata as different linetypes.

**Describe the solution you'd like**
It would be a nice feature to be able to choose whether one wants outcomes and strata to be represented by linetypes and colors or by colors and linetypes, respectively.
I can see they are in quosure form and they can be swapped using rlang quo_set_expr(), but this is only possible in cases where both multiple outcomes and multiple strata are present, as each mapping (linetype and color) is only generated when there is more than one value of the mapping. So it is not possible to e.g. make ggcuminc show multiple outcomes without strata as different colors instead of different linetypes.

### PR DESCRIPTION
**Pull request by ddsjoberg**
**What changes are proposed in this pull request?**
* By default, a model plot created with `ggsurvfit()` or `ggcuminc()` uses the color aesthetic to plot curves by the stratifying variable(s), and further, `ggcuminc()` uses the linetype aesthetic for plots that contain multiple outcomes (i.e. competing events). We not introduce the global option `"ggsurvfit.switch-color-linetype"` to switch these defaults, giving users more flexibility over the output figures. (#166)

**If there is an GitHub issue associated with this pull request, please provide link.**
closes #166

--------------------------------------------------------------------------------

Reviewer Checklist (if item does not apply, mark as complete)

- [x] Ensure all package dependencies are installed by running `renv::install()`
- [x] PR branch has pulled the most recent updates from master branch. Ensure the pull request branch and your local version match and both have the latest updates from the master branch.
- [x] If a new function was added, function included in `_pkgdown.yml`
- [x] If a bug was fixed, a unit test was added for the bug check
- [x] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [x] Overall code coverage remains >99.5%. Review coverage with `withr::with_envvar(new = c("NOT_CRAN" = "true"), covr::report())`. Begin in a fresh R session without any packages loaded. 
- [x] R CMD Check runs without errors, warnings, and notes
- [x] `usethis::use_spell_check()` runs with no spelling errors in documentation

When the branch is ready to be merged into master:
- [x] Update `NEWS.md` with the changes from this pull request under the heading "`# ggsurvfit (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [x] Increment the version number using `usethis::use_version(which = "dev")` 
- [x] Run `codemetar::write_codemeta()`
- [x] Run `usethis::use_spell_check()` again
- [x] Approve Pull Request
- [x] Merge the PR. Please use "Squash and merge".

### CODE REVIEW

**Review by psoldath — APPROVED** [review]
@ddsjoberg Thank you for implementing the request!

The code works perfect! When calling the global option options("ggsurvfit.switch-color-linetype" = TRUE), outcome and strata are switched like requested.

However, I don't know if I tested it correctly? I am very new to R, but I made it work by replacing the old code with the new code using the trace() function in the following functions:
trace(ggsurvfit:::.construct_censor_mark_aes, edit = TRUE)
trace(ggsurvfit:::.match_strata_level_to_color, edit = TRUE)
trace(ggsurvfit:::.construct_aes, edit = TRUE)
trace(ggsurvfit:::print.ggsurvfit, edit = TRUE)

**Review by psoldath — APPROVED** [review]

**Review by psoldath — APPROVED** [review]

**Comment by psoldath** [comment]
@ddsjoberg I download the pull request and the code works perfectly! Thank you for implementing the request.

**Comment by ddsjoberg** [comment]
Great! We'll merge this when @timtreis, a package co-author, has completed the review!

File: NEWS.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # ggsurvfit (development version)
 
+* By default, a model plot created with `ggsurvfit()` or `ggcuminc()` uses the color aesthetic to plot curves by the stratifying variable(s), and further, `ggcuminc()` uses the linetype aesthetic for plots that contain multiple outcomes (i.e. competing events). We not introduce the global option `"ggsurvfit.switch-color-linetype"` to switch these defaults, giving users more flexibility over the output figures. (#166)
```
Comment (timtreis) [inline NEWS.md:3]: We >now< introduce...

**Review by timtreis — CHANGES_REQUESTED** [review]
Minor typos, potentially unneccecary comments. Good example that illustrate functionality which work. Tests pass, but some paths are too long? Do you care within `ggsurvfit`?

```
   Found the following non-portable file paths:
     ggsurvfit/tests/testthat/_snaps/add_risktable_strata_symbol/add-risktable-strata-symbol-circle.new.svg
     ggsurvfit/tests/testthat/_snaps/add_risktable_strata_symbol/add-risktable-strata-symbol-default.new.svg
   
   Tarballs are only required to store paths of up to 100 bytes and cannot
   store those of more than 256 bytes, with restrictions including to 100
   bytes for the final component.
   See section ‘Package structure’ in the ‘Writing R Extensions’ manual.
   
   ```
   
Package Check works for me, vdiffr fails for some reason which I assume is Operating System + Version stuff, manually compared a large amount of snapshots and they were all the same.

File: NEWS.md (hunk 1 above)
Comment (ddsjoberg) [reply NEWS.md:3]: ahh, thank you! updated

File: R/add_risktable_strata_symbol.R [hunk 2]
```diff
@@ -91,7 +91,7 @@ update_add_risktable_strata_symbol <- function(p, add_risktable_strata_symbol_em
 
   # i hope this will work 100% of the time!
   # what modifications could a use make the the figures that could break this connection?
```
[the remark this answers is not in the document]
Comment (ddsjoberg) [reply R/add_risktable_strata_symbol.R:93]: oh yes, that is an oldie...no longer needed, good catch!

**Comment by ddsjoberg** [comment]
Thank you @timtreis for the thorough review!

vdiffr began failing everywhere (except locally) recently. I've since turned these off on GH Actions. I hope they begin working again. Previously, we had set to only run the vdiffr tests when the R version matched the current release of R and the check was being run on the same OS I develop on to avoid those kinds of failures on Linux.

I am sure if you looked at the differences in the failing vdiffr tests, there wouldn't be anything importantly different.

**Pull request merged**